django/db/models/query_utils.py

```diff
@@ -175,6 +175,26 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(self.identity)
 
+    @cached_property
+    def referenced_base_fields(self):
+        """
+        Retrieve all base fields referenced directly or through F expressions excluding
+        any fields referenced through joins.
+        """
+        # Avoid circular imports.
+        from django.db.models import F
+
+        expressions = list(self.flatten())
+        f_references = {expr.name for expr in expressions if isinstance(expr, F)}
+        q_references = {
+            child[0].split(LOOKUP_SEP, 1)[0]
```

Comment:
I saw from the linked PR that the `, 1` argument to `split` was a suggestion from Nick (to which I agree) but if we remove it, no test fail. Could you please add a case that could have multiple `LOOKUP_SEP` in it?

Reply:
I'm not sure I understand how we could add a test case for this, can you clarify what you have in mind?